### ISSUE TITLE
Feature Request: Evaluate Validation Set Using Generator 

### ISSUE THREAD

**Issue by wongjingping** [fchollet/keras#1632]
I noticed that there has yet to be an evaluation function implemented for validation data which uses a generator, from models.py line 1040:

` # TODO: call self.evaluate_generator()`

_Motivation_
Large validation datasets shouldn't be fully loaded into memory, only to be used at the end of an epoch (waste of memory). This feature would be quite useful in general for all applications.

_Implementation_
Evaluate the generator's next output by batches, with the user specifying an additional parameter in fit_generator to specify how many examples to evaluate from the validation set at the end of each epoch. Will add a function to the Model parent class so as to generalize to the Sequential and Graph classes.

Finally, as much as I've benefited much using Keras, I would also like to contribute back, hopefully by implementing this little feature successfully. As this is my first, would appreciate some guidance and any tips that you feel a beginner should know but might not!
p.s. I've read the CONTRIBUTING.md file :)

**Comment by pasky** [comment]
Please note that you can also use a callback to do this meanwhile:

````python
from keras.callback import Callback
class SampleValCB(Callback):
    def __init__(self, val):
        self.val = val

    def on_epoch_end(self, epoch, logs={}):
        mtloss = np.mean([self.model.test_on_batch(data) for data in sample_data(self.val, once=True)])
````

This assumes you have a generator ``sample_data`` that takes the dataset as first argument, and another argument once=True to indicate that it should go through the dataset exactly once.

**Comment by wongjingping** [comment]
@pasky Thanks for the suggestion; that's what I'm currently using. Wondering if that would be the 'proper way' to implement going forward?

**Comment by qdbp** [comment]
I'm working on something along these lines in #1758

**Comment by wongjingping** [comment]
@qdbp Thanks!
